src/emitter/batch_emitter.rs

```diff
@@ -224,7 +233,13 @@ impl Emitter for BatchEmitter {
     fn add(&mut self, payload: PayloadBuilder) -> Result<(), Error> {
         let batch = match self.event_store.lock() {
             Ok(mut store) => {
-                store.add(payload)?;
+                match store.add(payload) {
+                    Ok(_) => log::debug!("Added event to event store"),
+                    Err(e) => {
+                        log::warn!("Failed to add event to event store: {}", e);
```

Comment:
Nit: I think this has the potential to be an error because it could lead to data loss. On the other hand, we return an error from the function, so maybe that would be double error reporting?

Reply:
I think error level is probably more appropriate. I'm slightly conflicted about including it or not though as well. As you say it returns an Err which would be dealt with in user code, but I'm thinking it might be useful to be able to also see this occurring in a log after the fact, for potential diagnosis.